packages/daemon/src/context/ContextEvents.js

```diff
@@ -96,26 +130,32 @@ export default class ContextEvents {
         .pipe(
           filter((e: ContextEvent) => {
             return (
-              (e.type === 'contact_created' || e.type === 'contact_changed') &&
+              (e.type === 'contact_created' ||
+                (e.type === 'contact_changed' &&
+                  !this._publishingContacts.has(e.contact.localID))) &&
               e.contact.connectionState === 'sending'
             )
           }),
         )
         .subscribe(async (e: ContactCreatedEvent | ContactChangedEvent) => {
           const { contact, userID } = e
-          const user = this._context.openVault.identities.getOwnUser(
-            idType(userID),
-          )
-          const peer = this._context.openVault.identities.getPeerUser(
-            idType(contact.peerID),
-          )
+          const { identities } = this._context.openVault
+          const user = identities.getOwnUser(idType(userID))
+          const peer = identities.getPeerUser(idType(contact.peerID))
           if (!user || !peer) return
 
+          // Temp solution to handle possible situation where
+          // this event gets entered whilst already publishing
```

Comment:
I was having an issue with a race condition during contact creation where whilst the contact feed was being published for the first time another event was entering here triggered by `contact_changed` and then it was getting caught in an infinite loop. Me and Shane debugged this and decided this would be a quick fix. As it's only an issue before the contact has finished being created maybe we can restrict it to just that situation so not to affect successive triggers once the initial publish completes.

Reply:
Yes I'm mostly concerned about the fact it could prevent from publishing expected updates, and it would be hard to debug.
I think it's worth taking some time trying to solve the source of the issue, or making sure the change is restricted to this specific case.